From 4f95f7630828eddf7579c4f412e82eae6c1be2a7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sat, 11 Aug 2018 14:48:52 +0300
Subject: [PATCH] hl: make: fix {dali,gaudi}-*-build

On new systems the build path is yet to be created.
Therefore the {dali,gaudi}-*-build fails with:
$ make dali-asic-build
/bin/sh: 1: cannot create /home/dbarak/trees/npu-stack/buildroot-external/output/.hl-vars.mk: Directory nonexistent
Makefile:50: recipe for target 'hl-dali' failed
make: *** [hl-dali] Error 2

Fix by adding the output directory as a dependency for builds.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index e8488d8..c1dfeba 100644
--- a/Makefile
+++ b/Makefile
@@ -36,6 +36,9 @@
 
 export BR2_EXTERNAL := $(CURDIR)
 OUTPUT_DIR := $(BR2_EXTERNAL)/output
+$(OUTPUT_DIR):
+	@mkdir -p $@
+
 Makefile:;
 %:
 	@$(MAKE) -C ../buildroot O=$(OUTPUT_DIR) $@
@@ -68,8 +71,8 @@
 dali-onix-build: dali-onix-target
 dali-pldm-build: dali-pldm-target
 dali-asic-build: dali-asic-target
-gaudi-onix-build: gaudi-prepare
-dali-onix-build dali-pldm-build dali-asic-build: dali-prepare
+gaudi-onix-build: $(OUTPUT_DIR) gaudi-prepare
+dali-onix-build dali-pldm-build dali-asic-build: $(OUTPUT_DIR) dali-prepare
 dali-onix-build dali-pldm-build dali-asic-build gaudi-onix-build:
 	@$(MAKE) all
 
